src/audio/music.rs

```diff
@@ -119,13 +127,17 @@ impl Music {
     ///
     /// [`play`]: Music::play
     #[must_use]
-    pub fn from_memory(mem: &[u8]) -> Option<Music> {
-        let music_tmp =
-            unsafe { ffi::sfMusic_createFromMemory(mem.as_ptr() as *const _, mem.len()) };
+    pub fn from_memory(mem: impl Into<Box<[u8]>>) -> Option<Music> {
```

Comment:
To be honest, I would much rather stick to some kind of borrowing if possible, even if it involves lifetimes.
The original SFML API borrows here too, there is no keep-alive magic.

Reply:
It's probably feasible, however the InputStream is used in many places within the api, so designing the solution would take a bit longer, and I probably won't be able to test everything. Should I try it?

Reply:
That's actually kind of interesting. The Font and the Music are relying on having their data to be kept alive, but only the Music does have methods which start with open, Font, on the other hand, has a warning instead.

Reply:
Does InputStream need to be changed at all?
Music::from_stream takes &T, which can have a lifetime associated with it.
I made an experimental branch here:
https://github.com/jeremyletang/rust-sfml/tree/music-memsrc
Let me know what you think, and if it solves your case.

Reply:
Thanks, I'll see what I can do from here.

Reply:
There is still the need to pin an InputStream in a memory and keep it alive somehow for the lifetime of a Music. Box it and put in the field, perhaps?

Reply:
Is it required for soundness, or is it required for your use case because you want to store it along with the music?
If it's required for soundness, can you explain further? Because I think borrowing it should be enough for soundness.

If it's the latter, there could be a special Music type that stores the stream alongside the sfMusic handle.

Reply:
I think it's required, but not enough for soundness, as can be seen from documentation.
![image](https://user-images.githubusercontent.com/22716073/126012558-1efb4bcc-4fb0-465d-8fcf-9ded77238560.png)
There is certainly more to it due to STATUS_STACK_BUFFER_OVERRUN after calling sfMusic_destroy with the stream opened, so I don't want to insist on adding a box just for the sake of it, but it'd likely to become the necessary part of a fix.

Reply:
Alright, I understand why we need to keep the stream in memory now.

Sorry for taking so long to understand it. I'll think about how to best solve this, but I'll consider your OwnPtr solution too.